integration-tests/tests/src/tests/sync/app.ts

```diff
@@ -0,0 +1,322 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { ObjectId } from "bson";
+import { expect } from "chai";
+import { List } from "realm";
+import { importAppBefore } from "../../hooks";
+import { genPartition } from "../../utils/generators";
+import { TemplateReplacements } from "../../utils/import-app";
+
+const TestObjectSchema: Realm.ObjectSchema = {
+  name: "TestObject",
+  properties: {
+    doubleCol: "double",
+  },
+};
+
+const PersonForSyncSchema: Realm.ObjectSchema = {
+  name: "Person",
+  primaryKey: "_id",
+  properties: {
+    _id: "objectId",
+    age: "int",
+    dogs: "Dog[]",
+    firstName: "string",
+    lastName: "string",
+    realm_id: "string?",
+  },
+};
+
+const DogForSyncSchema: Realm.ObjectSchema = {
+  name: "Dog",
+  primaryKey: "_id",
+  properties: {
+    _id: "objectId",
+    breed: "string?",
+    name: "string",
+    realm_id: "string?",
+  },
+};
+
+interface IPersonForSyncSchema {
+  _id: ObjectId;
+  age: number;
+  dogs: IDogForSyncSchema[];
+  firstName: string;
+  lastName: string;
+  realm_id: string | undefined;
+}
+
+interface IDogForSyncSchema {
+  _id: ObjectId;
+  breed: string | undefined;
+  name: string;
+  realm_id: string | undefined;
+}
+
+describe("App", () => {
+  describe("instantiation", function () {
+    type ConfigContext = {
+      conf: { id: string; baseUrl: string };
+    } & Mocha.Context;
+    afterEach(async () => {
+      Realm.clearTestState();
+    });
+    before(async function (this: ConfigContext) {
+      this.conf = { id: "smurf", baseUrl: "http://localhost:9090" };
+    });
```

Comment:
I don't think this is worth the mental overhead - it's simpler to just inline these values when used below.

```suggestion
    afterEach(async () => {
      Realm.clearTestState();
    });
```